WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.swift

```diff
@@ -0,0 +1,1047 @@
+    
+    // MARK: - Table View Handling
+    
+    func tableView(tableView: UITableView!, didSelectRowAtIndexPath indexPath: NSIndexPath!) {
+        assert(false, "You should implement this method in the subclass")
+    }
+    
+    func tableViewDidChangeContent(tableView: UITableView!) {
+        // After any change, make sure that the no results view is properly
+        // configured.
+        
+        configureNoResultsView()
+    }
+    
+    func tableView(tableView: UITableView!, willDisplayCell cell: UITableViewCell!, forRowAtIndexPath indexPath: NSIndexPath!) {
+        guard isViewOnScreen() && !isSearching() else {
+            return
+        }
+        
+        // Are we approaching the end of the table?
+        if indexPath.section + 1 == tableView.numberOfSections
+            && indexPath.row + self.dynamicType.postsLoadMoreThreshold >= tableView.numberOfRowsInSection(indexPath.section) {
+            
+            // Only 3 rows till the end of table
+            if currentPostListFilter().hasMore {
+                syncHelper.syncMoreContent()
+            }
+        }
+    }
+    
+    func configureCell(cell: UITableViewCell, atIndexPath indexPath: NSIndexPath) {
+        assert(false, "You should implement this method in the subclass")
+    }
+    
+    // MARK: - Actions
+    
+    @IBAction func refresh(sender: AnyObject) {
+        syncItemsWithUserInteraction(true)
+        
+        WPAnalytics.track(.PostListPullToRefresh, withProperties: propertiesForAnalytics())
+    }
+    
+    @IBAction func handleAddButtonTapped(sender: AnyObject) {
+        createPost()
+    }
+    
+    @IBAction func handleSearchButtonTapped(sender: AnyObject) {
+        toggleSearch()
+    }
+    
+    @IBAction func didTapNoResultsView(noResultsView: WPNoResultsView) {
+        WPAnalytics.track(.PostListNoResultsButtonPressed, withProperties: propertiesForAnalytics())
+        
+        if currentPostListFilter().filterType == .Scheduled {
+            let index = indexForFilterWithType(.Draft)
+            setCurrentFilterIndex(index)
+            return
+        }
+        
+        createPost()
+    }
+    
+    @IBAction func didTapFilterButton(sender: AnyObject) {
+        displayFilters()
+    }
+
+    // MARK: - Synching
+
+    func automaticallySyncIfAppropriate() {
+        // Only automatically refresh if the view is loaded and visible on the screen
+        if isViewLoaded() == false || view.window == nil {
+            DDLogSwift.logVerbose("View is not visible and will not check for auto refresh.")
+            return
+        }
+
+        // Do not start auto-sync if connection is down
+        let appDelegate = WordPressAppDelegate.sharedInstance()
+        
+        if appDelegate.connectionAvailable == false {
+            configureNoResultsView()
+            return
+        }
+
+        if let lastSynced = lastSyncDate() where abs(lastSynced.timeIntervalSinceNow) > self.dynamicType.postsControllerRefreshInterval {
```

Comment:
@kurzee - I believe this is the cause of the bug you were seeing.

Nice catch @SergioEstevao!